File: app/src/main/java/com/hamza/ieeechallenge/MainActivity.java
package com.hamza.ieeechallenge;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.google.firebase.auth.FirebaseAuth;
import com.hamza.ieeechallenge.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;
    private FirebaseAuth firebaseAuth;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        firebaseAuth = FirebaseAuth.getInstance();

        binding.btnSignOut.setOnClickListener(view1 -> {
            firebaseAuth.signOut();
            startActivity(new Intent(MainActivity.this , SendOtpActivity.class));
        });
    }
}

File: app/src/main/java/com/hamza/ieeechallenge/ui/MainActivity.kt
package com.hamza.ieeechallenge.ui

import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.PhoneAuthOptions
import com.hamza.ieeechallenge.R
import com.hamza.ieeechallenge.ui.auth.AuthViewModel
import dagger.hilt.android.AndroidEntryPoint
import java.util.concurrent.TimeUnit
import javax.inject.Inject


@AndroidEntryPoint
class MainActivity @Inject constructor(): AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
}